### DISCUSSION TITLE
How to show Order Total on last page in birt report.

### DISCUSSION THREAD

**Discussion by sharath492** [eclipse-birt/birt#1105] [Q&A]
Hi all,

I am using Birt report version 3.2.23.

On multiple pages need to show only Order Total on last page. For previous pages need to populate as 'Continued'. And the footer text should always shown on all pages.

Can anyone please help on this
[atkorepdf-ukpur
![Capture](https://user-images.githubusercontent.com/41218746/200516166-83dad85a-56c9-4c66-b234-70902c790ef3.JPG)
chaseorderfromb2bexml.zip](https://github.com/eclipse/birt/files/9959232/atkorepdf-ukpurchaseorderfromb2bexml.zip)
?

Thanks& regards,
Sharath

**Comment by speckyspooky** [comment]
Hello sharath492,

welcome back.

I have added an example of an implementation option. You will find the according "Order Total" on your MasterPage.
The dynamic-text-element has a very small script in the method "onRender()".
To solve the requirement of total only on last page I used a compare of pageNumber and totalPage.

Please be aware that the value of "totalPage"-variable is only on on Render-Task correct set.

**example of summary table & script:**
![2022-11-08 20_53_14-Greenshot Editor](https://user-images.githubusercontent.com/41593722/200661918-e56adc35-4b90-49bf-ae56-c92d921c0e69.png)


**example of summary table:**
[rexelaccpaypdf-nlpurchaseorderfromb2bexml_birt_48.pdf](https://github.com/eclipse/birt/files/9964648/rexelaccpaypdf-nlpurchaseorderfromb2bexml_birt_48.pdf)

**example report:**
[rexelaccpaypdf-nlpurchaseorderfromb2bexml-order-total.zip](https://github.com/eclipse/birt/files/9964650/rexelaccpaypdf-nlpurchaseorderfromb2bexml-order-total.zip)

**Reply by sharath492** [reply]
Hello [speckyspooky](https://github.com/speckyspooky),

I have tried above your example but not getting output. May i know where is pageNumber and totalPage are created?

And i have tried another type but that also not getting.

Could you please take a look on my attachments.
[atkorepdf-uk_XR.zip](https://github.com/eclipse/birt/files/9967954/atkorepdf-uk_XR.zip)

**Reply by speckyspooky** [reply]
Hello sharath492,
I have added the script at the dynamic-text-element of the "total value" which is located into the grid of the master-page-layout.
If you mark the element go to script page and select the method "onRender" you wil see my code.

I will take a look into your report.

**Reply by speckyspooky** [reply]
I checked your report. The I removed the script on the "onPageEnd" and added 4 lines on your onRender-script.
So I get your required result

 **Added code on your onRender-script** 

this.text = "Continued";
if ( pageNumber == totalPage ) {
	this.text = Formatter.format(reportContext.getPageVariable("totalValue"), "#,###.00");
}
if(this.text == "Continued")
{
this.getStyle().fontSize = "10";
}
else
this.getStyle().fontSize = "22";

**Reply by sharath492** [reply]
Hello [speckyspooky](https://github.com/speckyspooky),

i tried your onRender script but in both the pages displaying OrderTotal value but it should be display continued in 1st page and last page should be OrderTotal.

btw  pageNumber and totalPage is variables? if not where we are creating those?

**Reply by speckyspooky** [reply]
Both variables are default-variables which exists automatically in the BIRT-report-context.
There is no task or step to define these.

The problem is that you I changed your report on a second position. I removed your variable-setting.
Attached I send you your report with my 2 changes and also the PDF-output.

I have tested the report with BIRT 4.6 and BIRT 4.8.

You use also the BIRT 4.6 environment with the report version 3.2.23 according to your report and so it should be work.

[tg_atkorepdf-ukpurchaseorderfromb2bexml.zip](https://github.com/eclipse/birt/files/9984497/tg_atkorepdf-ukpurchaseorderfromb2bexml.zip)

[atkorepdf-ukpurchaseorderfromb2bexml.pdf](https://github.com/eclipse/birt/files/9984503/atkorepdf-ukpurchaseorderfromb2bexml.pdf)

Best regards
Thomas

**Reply by TheLaserNinja** [reply]
Are you viewing it in BIRT's preview, or the BIRT Viewer?

**Reply by sharath492** [reply]
Hello [speckyspooky](https://github.com/speckyspooky),

I have downloaded your rpt design attachment and previewed and passed value for TotalOrderAmount but not getting 'Continued' on 1st page.
![Capture](https://user-images.githubusercontent.com/41218746/202449739-4994c238-2379-494d-8713-a845293aadb3.JPG)
[atkorepdf-ukpurchaseorderfromb2bexml.zip](https://github.com/eclipse/birt/files/10031593/atkorepdf-ukpurchaseorderfromb2bexml.zip)

**Reply by speckyspooky** [reply]
I think I know the difference between your result and my result.

I use BIRT 4.6 - create the HTML preview and started from there the PDF: result is perfect
I use BIRT 4.6 - create the PDF-directly: result is according to your result

Please start the PDF-generation by using the HTML preview and check the result.

The process handling of BIRT is different between direct starting PDF from eclipse or starting view HTML-preview to generate the PDF.
The reason is the RUN.Task+RENDER.Task is needed the direct call starts directly the RENDER.Task but special features need the RUN.Task

Best regards

**Reply by hvbtup** [reply]
Yes, generally, "page variables" need separate RunTask and RenderTask.
